doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,586 @@
+│   └───Available nodes: primary, render
+├───Framebuffer size
+│   ├───Width: [0, 16384]
+│   └───Height: [0, 16384]
+├───Connectors
+|   |   ...
+│   ├───Connector 3
+│   ├───Object ID: 101
+│   ├───Type: HDMI-A
+│   ├───Status: connected
+│   ├───Physical size: 530x300 mm
+│   ├───Subpixel: unknown
+│   ├───Encoders: {3}
+│   ├───Modes
+│   │   ├───1920x1080@60.00 preferred driver phsync pvsync 
+│   │   ├───1920x1080@60.00 driver phsync pvsync 16:9 
+│   │   ├───...
+|   |   ...
+├───Encoders
+│   ├───Encoder 0
+│   │   ├───Object ID: 82
+│   │   ├───Type: TMDS
+│   │   ├───CRTCS: {0, 1, 2, 3}
+│   │   └───Clones: {0}
+|   |   ...
+├───CRTCs
+│   ├───CRTC 0
+│   │   ├───Object ID: 72
+│   │   ├───Legacy info
+│   │   │   ├───Mode: 1920x1080@60.00 preferred driver phsync pvsync 
+│   │   │   └───Gamma size: 256
+|   |   ...
+└───Planes
+    ├───Plane 0
+    │   ├───Object ID: 40
+    │   ├───CRTCs: {3}
+    │   ├───Legacy info
+    │   │   ├───FB ID: 0
+    |   ...
+```
+You want to make sure that you have:
+- 1 Node
+- 1 or more Connector(s)
+- 1 or mor CRTC(s)
+- 1 or more Plane(s)
+
+Finally, If you have a display connected, then one of your `Connector`s should read `Status: connected`. This `Connector` will also list many `Modes` that
+the  display can use.
+
+#### Troubleshooting
+If `drm_info` does *not* produce results similar to those above, then this
+indicates that your kernel **is NOT yet ready** to support the GPU. This
+means that kernel enablement is required before Ubuntu Frame can be enabled.
+
+### 3. Driver Support Check
+Finally, we will need to check that you have the proper userspace drivers
+installed on the system, namely:
+- `EGL` (`libEGL.so.1`)
+- `GLES` (`libGLESv2.so.2`) 
+- `GBM` (`libgbm.so.1`)
+
+While *Mir* can be enabled on platforms that do not support `GBM/KMS`, this
+requires a paltform implementation to be written in *Mir*. This work is out
+of the scope of this guide.
+
+## How to assemble a `graphics-core22` provider snap
+With the prerequisites out of the way, it is time to assemble a `graphics-core22` snap.
+
+The snap itself must provide the libraries described in 
+[this document](https://mir-server.io/docs/the-graphics-core22-snap-interface#heading--lists--supported-apis).
+Unless you are incredibly space-constrained or in an unusual circumstance, it is a good idea to
+provide **all** of these libraries. Even if your board doesn’t support these libraries, a failure to include
+them will result in applications that would otherwise fall back to a supported API failing to start.
+
+
+`libEGL`, `libGLESv2` and `libgbm` should be provided by the vendor's drivers.
+`libvulkan` *may* be provided by the vendor as well.
+The remaining packages should be available in the Ubuntu archive.
+
+Most of these libraries do **NOT** need to be in a fixed location in the snap.
+However, the following paths **MUST** be provided in a fixed location by your snap:
+
+
+1. libdrm:
+    ```yaml
+    parts:
+      drm-i386:
```

Comment:
The `-i386` suffix is specific to supporting i386 on x86_64 - this should be noted in a separate section I think, pointing at the Mesa implementation, as an optional feature.